src/cmakeWorkspaceFolder.ts

```diff
@@ -214,30 +53,85 @@ export class CMakeWorkspaceFolderController implements vscode.Disposable {
      * - Where we search for variants
      * - Where we search for workspace-local kits
      */
-    private _activeFolder?: CMakeWorkspaceFolder;
-    get activeFolder() {
-        return this._activeFolder;
+
+    get activeFolderPath(): string | undefined {
+        return this.activeCMakeProject?.folderPath;
+    }
+    /**
+     * The name of the folder for this CMakeProject instance
+     */
+    get activeFolderName(): string | undefined {
+        return this.activeCMakeProject?.folderName;
     }
-    setActiveFolder(ws: vscode.WorkspaceFolder | undefined) {
-        if (ws) {
-            this._activeFolder = this.get(ws);
-        } else {
-            this._activeFolder = undefined;
+
+    private activeCMakeProject: CMakeProject | undefined;
+    setActiveCMakeProject(workspaceFolder?: vscode.WorkspaceFolder, openEditor?: vscode.TextEditor, folderName?: string): CMakeProject | undefined {
+        if (!workspaceFolder && !openEditor && !folderName) {
+            return this.activeCMakeProject;
+        }
+        // Setting the active CMake project from the status bar
```

Comment:
I'm confused as to why there is a reference to the status bar here.  It seems like the open editor should have precedence. But I'm wondering if the decision making should be done here.  It doesn't appear to take into account the setting to not change the active project when the openEditor changes.  I think `extension.ts` is a better place to do that.  And then this function is just a simple 'set' like the old `setActiveFolder` function was.

Reply:
I had to update the comment, it was not very clear. The decision for the status bar is being made in the extension.ts

```typescript
        // When the folderName is defined, it means the active CMake project is being updated from the status bar.
        // In this case, we need to search for the new project in the list of all projects.
        if (!workspaceFolder && !openEditor && folderName) {
```